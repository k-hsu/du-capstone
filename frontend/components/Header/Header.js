import React from "react";
import Flex from "../Flex/Flex";
import Image from "../Image/Image";
import Link from "../Link/Link";
import Text from "../Text/Text";
import { spacing } from "../../theme";

const Header = (props) => {
  return (
    <Flex
      p={`${spacing["0.75"]} ${spacing["4"]}`}
      gap={spacing["0.75"]}
      {...props}
    >
      <Link href="/">
        <Flex alignItems="center" height="100%">
          <Image
            src="/DU-Logo-Mark.svg"
            alt="capstone-logo"
            width="64px"
            height="64px"
          />
        </Flex>
      </Link>
      <Text as="h1" lineHeight={spacing["1.5"]}>
        Kobi&apos;s Capstone Project
      </Text>
    </Flex>
  );
};

export default Header;
