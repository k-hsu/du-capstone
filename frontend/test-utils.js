import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { renderHook } from "@testing-library/react-hooks";

function setup(jsx) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}

export * from "@testing-library/react";
export { renderHook, setup };
