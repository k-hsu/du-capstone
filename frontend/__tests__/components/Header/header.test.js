import React from "react";
import { render, screen } from "../../../test-utils";
import Header from "../../../components/Header/Header";

describe("Header", () => {
  it("should render header component", () => {
    render(<Header />);
    expect(screen.getByAltText("capstone-logo")).toBeInTheDocument();
    expect(screen.getByText("Kobi's Capstone Project")).toBeInTheDocument();
  });
});
