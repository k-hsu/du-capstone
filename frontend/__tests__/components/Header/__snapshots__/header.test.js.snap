// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render header component styling 1`] = `
<div
  className="css-15dhzng"
  direction="row"
>
  <span
    style={
      {
        "background": "none",
        "border": 0,
        "boxSizing": "border-box",
        "display": "inline-block",
        "height": "initial",
        "margin": 0,
        "maxWidth": "100%",
        "opacity": 1,
        "overflow": "hidden",
        "padding": 0,
        "position": "relative",
        "width": "initial",
      }
    }
  >
    <span
      style={
        {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "padding": 0,
          "width": "initial",
        }
      }
    >
      <img
        alt=""
        aria-hidden={true}
        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2732%27%20height=%2732%27/%3e"
        style={
          {
            "background": "none",
            "border": 0,
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      />
    </span>
    <img
      alt="capstone-logo"
      data-nimg="intrinsic"
      decoding="async"
      onError={[Function]}
      onLoad={[Function]}
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style={
        {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
    <noscript>
      <img
        alt="capstone-logo"
        data-nimg="intrinsic"
        decoding="async"
        loading="lazy"
        src="/DU-Logo-Mark.svg"
        srcSet="/DU-Logo-Mark.svg 1x, /DU-Logo-Mark.svg 2x"
        style={
          {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </noscript>
  </span>
  <text
    className="css-1kjt1be"
    fontFamily="IBM Plex Sans"
  >
    Kobi's Capstone Project
  </text>
</div>
`;
